components/TextEditor/TextEditorContainer.tsx

```diff
@@ -24,29 +22,41 @@ const TextEditorContainer: FunctionComponent<TextEditorContainerProps> = ({heigh
     const [error, setError] = useState("");
     const getEditorContentIfMountedRef: MutableRefObject<EditorContentGetter> = useRef(() => "");
     const router=useRouter();
-    
+
     function handleIsEditorMounted(_getEditorContents: EditorContentGetter): void {
         setIsEditorReady(true);
+        if(isEditorReady) {
         getEditorContentIfMountedRef.current = _getEditorContents;
-     }
+     }}
 
-   
-    function submitForm() {
-        const testAnswer=getEditorContentIfMountedRef.current();
-        addTestSubmisson(token,{testId,testAnswer})
-            .then((response)=>{  
+
+    function submitForm(): void {
+        const testAnswer = getEditorContentIfMountedRef.current();
+        addTestSubmission(token,{testId,testAnswer})
+            .then((response)=>{
                 if (response.ok) {
-                    router.push(`/submitted?token=${token}`);
+                    router.push('/testlibrary').then(r => r.valueOf());
                 } else {
                      throw Error(response.statusText);
                 }
-            })           
+            })
             .catch(error=>{console.log(error);
                 setError("There was an error submitting your test")});
-            
     }
-
+    let link = "";
+    let as = "";
+    const handleSubmit = () => {
+        submitForm();
+        if(TestList.length-1 > useStatus().length) {
+            link = "/testlibrary";
+            as = "testlibrary"
+        } else {
+            link = "/submitted";
+            as = "thankyou"
+        }
+    }
```

Comment:
I _think_ this is all handled by the testLibrary page, so fine to just link to the testlibrary.